### ISSUE TITLE
Expand quick switcher to also search in projects functions/articles

### ISSUE THREAD

**Issue by martinklepsch** [cljdoc/cljdoc#194]
We have [this quick switcher](https://imgur.com/a/8Z1bwm5) and I think it would be really amazing if it could also work to search inside a project. 

To do this the following would be necessary: 

- [ ] Provide some JSON "table of contents" to the JS-based search. This could be an XHR request or embedded into the page when it's rendered. (This would be similar to the quick switcher which retrieves candidates from local storage.)
- [ ] Establish a second state for the quick switcher that is "scoped" to the current project. I imagine that this being the default when bringing it up. Perhaps you could get to the global state by hitting backspace while the input is empty. 

I've seen examples of this in the wild but can't find something immediately. I'll eventually add a mockup to this issue. 

By having clear scoping we avoid the issue of global search (for now) and only offer two specific modes:

- global search (results are artifacts)
- local search (results are namespaces, functions, articles, ...)

... which I think is a good compromise and covers the most common usage patterns.

**Comment by n2o** [comment]
Hi, thanks for your effort in cljdoc!

I'd like to focus a bit more on this issue, because it is really annoying if a project's documentation is only partly usable, because of the lack of a search engine searching inside the article. For example, the documentation of [reitit](https://cljdoc.org/d/metosin/reitit/0.5.5/doc/introduction) is really good and moved to cljdoc-only last week, but it is hard to navigate cljdoc without the option to search or jump to passages.

I'd love to see this feature in this project, and maybe I find the time to support you with some lines of code to implement it / help you with it :-)

**Comment by martinklepsch** [comment]
Yes please, that would be fantastic! Let me when you have any questions :)
